apps/site/src/app/portal/Portal.tsx

```diff
@@ -1,21 +1,50 @@
-import api from "@/lib/utils/api";
+import { redirect } from "next/navigation";
+import type { Metadata } from "next";
 
-async function getIdentity(): Promise<string> {
-	const res = await api.get<string>("/user/me");
-	return res.data;
+import VerticalTimeline from "./VerticalTimeline";
+import Message from "./Message";
+import getUserIdentity from "@/lib/utils/getUserIdentity";
+
+import styles from "./Portal.module.scss";
+
+export const enum PortalStatus {
+	pending = "PENDING_REVIEW",
+	reviewed = "REVIEWED",
+	accepted = "ACCEPTED",
+	rejected = "REJECTED",
+	waitlisted = "WAITLISTED",
+	confirmed = "CONFIRMED",
 }
 
+export const metadata: Metadata = {
+	title: "Portal | IrvineHacks 2024",
+};
+
 async function Portal() {
-	const identity = await getIdentity();
+	const identity = await getUserIdentity();
+	const status = identity.status;
+
+	if (status === null) {
+		redirect("/apply");
+	}
+
 	return (
-		<div className="min-h-screen flex flex-col justify-center bg-white text-black">
-			<h1>Hello</h1>
-			{Object.entries(identity).map(([key, value]) => (
-				<p key={key}>
-					{key} - {value}
-				</p>
-			))}
-		</div>
+		<>
+			<section className=" w-full flex items-center flex-col">
+				<div className="m-24">
+					<h1
+						className={`${styles.title} font-display sm:text-[3rem] text-[#fffce2] text-6xl text-center`}
+					>
+						Portal
+					</h1>
+				</div>
+				<div className="bg-white text-black max-w-4xl rounded-2xl p-6 flex flex-col mb-24 w-full">
+					<h2 className="text-4xl font-semibold">Status</h2>
+					<VerticalTimeline status={status} />
+					<Message status={status as PortalStatus} />
```

Comment:
Question: is there a particular reason the cast `as PortalStatus` was needed?

Reply:
I believe it is to avoid a typescript error: 
`Type 'string' is not assignable to type 'PortalStatus'.ts(2322)`

Reply:
Got it, we can revisit this: think we could have identity itself type the status.